File: 백준/Silver/1475. 방 번호/방 번호.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    public static void main(String[] args) throws IOException {
        String N = br.readLine();
        int[] arr = new int[10];

        // 6이랑 9일때만 특수케이스로
        for (int i = 0; i < N.length(); i++) {
            int num = N.charAt(i) - '0';
            if (num == 9 || num == 6) {
                int small = (arr[9] >= arr[6]) ? 6 : 9;
                arr[small]++;
            } else {
                arr[num]++;
            }
        }

        int max = -1;
        for (int i = 0; i <= 9; i++) {
            max = Math.max(max, arr[i]);
        }
        bw.write(max + "\n");
        bw.flush();
        bw.close();
    }
}

File: 백준/Silver/1475. 방 번호/방 번호.py
import math

N = input()

numbers = [0] * 10

for i in range(len(N)):
    if int(N[i]) == 9: # 9일때 6으로 몰빵
        numbers[6] += 1
    else:
        numbers[int(N[i])] += 1

max_val = 0
numbers[6] = math.ceil(numbers[6] / 2)

for num in numbers:
    max_val = max(num, max_val)
print(max_val)